include/fastaq_handler.h

```diff
@@ -4,19 +4,27 @@
 #include <string>
 #include <cstdint>
 #include <fstream>
+#include <exception>
 #include <boost/iostreams/filtering_streambuf.hpp>
 #include <boost/log/core.hpp>
 #include <boost/log/trivial.hpp>
 #include <boost/log/expressions.hpp>
+#include <cstdio>
+#include <zlib.h>
+#include "kseq.h"
+KSEQ_INIT(gzFile, gzread)
 
 namespace logging = boost::log;
 
 struct FastaqHandler {
-    bool gzipped;
-    std::ifstream fastaq_file;
-    boost::iostreams::filtering_istreambuf inbuf;
-    std::istream instream;
-    std::string line;
+private:
+    int closed_status; // see
+                       // https://github.com/attractivechaos/klib/blob/928581a78413bed4efa956731b35b18a638f20f3/kseq.h#L171
```

Comment:
Is this documentation correct? I think `closed_status` refers to return values of `gzclose()`, not return values of `kseq_read()`

Reply:
Yeah, not sure what happened here... :confused: